{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Beijing Lineage Phylogeny Construction - Run 4 ##\n",
    "\n",
    "### 3 July 2016 ###\n",
    "\n",
    "The input data for this run are VCF files containing the SNPs from 110 unique strains from ~150 samples from the Beijing Lineage, where SNPs from repetitive regions and antibiotic-resistance were removed along with those whose frequency was less than 2. The entire list of strains used in this run can be found in the convertvcf folder in the file unique_strains.txt.\n",
    "\n",
    "#### VCF Conversion ####\n",
    "\n",
    "The collection of VCF files were converted to FASTA file format with the tool convertvcf.py under the convertvcf folder with the following commands:\n",
    "\n",
    "```bash\n",
    "run=run4\n",
    "\n",
    "inputDir=/home/zhf615/TB_test/MALAWI/test/shuffled/results\n",
    "outputDir=/global/scratch/seanla/Data/MTBC/$run\n",
    "snpDir=$outputDir/snps\n",
    "snpPrefix=$snpDir/$run\n",
    "\n",
    "convertVcfDir=/home/seanla/Projects/beijing_ancestor_mtbc/convertvcf\n",
    "strains=$convertVcfDir/unique_strains.txt\n",
    "convertVcf=$convertVcfDir/convertvcf.py\n",
    "\n",
    "mkdir -p $snpDir\n",
    "\n",
    "mkdir -p $phymlDir\n",
    "\n",
    "python $convertVcf -i $inputDir -o $snpPrefix -s $strains -r \"(\\d|\\D)*_final.vcf$\" -f -p\n",
    "```\n",
    "\n",
    "Where `(\\d|\\D)*_final.vcf$` is the regular expression for the appropriate VCF files. Processing revealed that there are 7806 sites (i.e. the length of each strain sequence is 7806 bases).\n",
    "\n",
    "#### Best Fit Model of DNA Evolution - jModelTest ####\n",
    "\n",
    "jModelTest was used to infer the best-fit model of nucleotide evolution using the Akaike information criterion with the following commands:\n",
    "\n",
    "```bash\n",
    "run=run4\n",
    "\n",
    "prefix=/global/scratch/seanla/Data/MTBC/$run\n",
    "phylip=$prefix/snps/$run.phy\n",
    "jmodelDir=$prefix/jmodeltest\n",
    "output=$jmodelDir/jmodeltest-results.out\n",
    "jModelTest=/home/seanla/Software/jmodeltest-2.1.10/jModelTest.jar\n",
    "\n",
    "mkdir -p $jmodelDir\n",
    "\n",
    "java -jar $jModelTest -tr ${PBS_NUM_PPN} -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "An explanation of the parameters used is as follows:\n",
    "* -AIC indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* -a indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* -f indicates we included models with unequal base frequencies.\n",
    "* -g 4 indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* -i indicates we included models with a proportion invariable sites.\n",
    "* -H AIC indicates we used the AIC information criterion for clustering search. (jModelTest indicated that this option has no effect for our dataset.)\n",
    "\n",
    "jModelTest ouputted GTR+G (General Time Reversible model with rate variation among sites) as the best fit model for our data.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+G           0.16    0.35    0.32    0.17    0.00    0.00      0.901   3.626   0.431   0.517   2.731   1.000 N/A        0.59\n",
    "```\n",
    "\n",
    "SNP alignments in the PHYLIP format were fed to jModelTest.\n",
    "\n",
    "#### Neighbor-joining Tree with MEGA ####\n",
    "\n",
    "The neighbor-joining tree was constructed with MEGA with the following parameters:\n",
    "\n",
    "* Test of phylogeny was set to bootstrap with 1000 pseudoreplicates.\n",
    "* Substitution model/method was set to no. of differences.\n",
    "* Substitutions to include were transitions and transversions.\n",
    "* Rates among sites was set to Gamma distributed with a Gamma parameter of 0.59 (as per the output of jModelTest)\n",
    "\n",
    "SNP alignments were fed to MEGA in the FASTA format.\n",
    "\n",
    "#### Maximum-likelihood Tree with MEGA ####\n",
    "\n",
    "The maximum-likelihood tree was constructed using MEGA with the following parameters:\n",
    "\n",
    "* Test of phylogeny was set to bootstrap with 1000 pseudoreplicates.\n",
    "* The model of substitution was set to General Time Reversible model.\n",
    "* Rates among Sites was set to Gamma Distributed.\n",
    "* No. of discrete gamma categories was set to 4 (as per the default number of categories in jModelTest)\n",
    "* The ML heuristic used was Nearest-Neighbor-Interchange (NNI)\n",
    "* The initial tree for ML was created automatically with default being neighbor-joining/Bio neighbor-joining\n",
    "* Branch swap filter was set to none.\n",
    "* Number of threads used was 12.\n",
    "\n",
    "SNP alignments were fed to MEGA in the FASTA format.\n",
    "\n",
    "#### Bayesian Inference with BEAST ####\n",
    "\n",
    "The optimal tree distribution was found using Bayesian inference with the program BEAST with the following parameters:\n",
    "\n",
    "##### Site Model #####\n",
    "* The Gamma category count was set to 4.\n",
    "* The Gamma shape parameter was set to 0.59.\n",
    "* The proportion invariant (proportion of invariant sites) was set to 0.0.\n",
    "* The substitution model was set to GTR, with each substitution parameter (AC, AG, etc) set to 1.0. (Default value)\n",
    "* All substitution parameters, with the exception of CT, were estimated. (Default value)\n",
    "* Frequencies for each were estimated. (Default value)\n",
    "\n",
    "##### Clock Model #####\n",
    "* A strict clock was used with a clock rate of 1.0.\n",
    "\n",
    "##### Priors #####\n",
    "* The tree prior was set to the Yule Model.\n",
    "* Birth rate was set to uniform, with an initial value of 1.0 and constrained within the real numbers ([-inf, inf]).\n",
    "* The GTR substitution parameters was set to a Gamma distribution.\n",
    "* The GTR substitution parameters for each substitution type was initially set to 1.0 and constrained within the positive real numbers ([0,inf]).\n",
    "\n",
    "##### MCMC #####\n",
    "* The chain length was set to 10 000 000.\n",
    "* The number of pre-burnin was set to 0.\n",
    "* The number of initialization attempts was set to 10.\n",
    "* Sample from prior was not initialized.\n",
    "\n",
    "SNP alignments were given to BEAST in FASTA file format.\n",
    "\n",
    "### 5 July 2016 ###\n",
    "\n",
    "The Neighbor-Joining tree was visualized by passing the NEWICK file `run4mega-nj.nwk` to Dendroscope (the extension was changed to `newick` so that Dendroscope could accept it), where the labels of the internal nodes were interpreted as bootstrap values.\n",
    "\n",
    "![Neighbor-joining Tree](run4mega-nj.png)\n",
    "\n",
    "The Bayesian consensus tree was visualized using the following steps:\n",
    "* TreeAnnotator v2.4.2 was used first to find the consensus topology, where the burnin percentage was first set to 10%. The node heights were set to mean heights, and the target tree was set to maximum clade credibility tree.\n",
    "* The output file `run4bayes-consensus.nexus` given by TreeAnnotator was then fed into FigTree and the option 'node labels' was set.\n",
    "\n",
    "![Bayesian Consensus Tree](run4bayes-consensus.nexus.png)\n",
    "\n",
    "### 7 July 2016 ###\n",
    "\n",
    "RAxML was used to construct the maximum likelihood tree and perform 1000 rapid bootstraps using the following command:\n",
    "\n",
    "```bash\n",
    "run=run4\n",
    "dir=/global/scratch/seanla/Data/MTBC/$run\n",
    "input=$dir/snps/$run.phy\n",
    "outputdir=$dir/raxml\n",
    "output=${run}raxml\n",
    "\n",
    "mkdir -p $outputdir\n",
    "\n",
    "raxml -T ${PBS_NUM_PPN} -f a -x 61938 -p 889579 -# 1000 -k -m GTRGAMMA -s $input -w $outputdir -n $output\n",
    "```\n",
    "\n",
    "An explanation of the parameters follows:\n",
    "* `-f a` indicates we performed a rapid bootstrap analysis and search for best-scoring ML tree in one program run.\n",
    "* `-x 61938` indicates we gave the integer 61938 as a random seed to initialize rapid bootstrapping.\n",
    "* `-p 889579` indicates we passed the integer 889579 as a random seed to initialize parsimony inferences. This is necessary when using any algorithm/option that requires randomization.\n",
    "* `-# 1000` indicates we performed 1000 bootstrap pseudoreplicates (this is according to the [website](http://sco.h-its.org/exelixis/web/software/raxml/hands_on.html) of the developers).\n",
    "* `-k` indicates that bootstrapped trees are printed with branch lengths.\n",
    "* `-m GTRGAMMA` indicates the the model of nucleotide substitution is GTR with optimization of substitution rates and the usage of a gamma model for rate heterogeneity with an estimated alpha parameter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
