{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "** 22 June 2016 **\n",
    "\n",
    "The 154 samples from the Beijing lineage were filtered to remove redundant samples from the same lineage, arriving to a filtered list of 110 unique strains. The list of removed samples These unique strains were fed into `convertvcf.py` to construct FASTA and PHYLIP files. This was done with the following commands:\n",
    "\n",
    "```\n",
    "$ run=run2\n",
    "$ inputDir=/home/zhf615/TB_test/MALAWI/test/shuffled/results\n",
    "$ outputPrefix=/global/scratch/seanla/Data/MTBC/$run/snps/$run\n",
    "$ convertVcfDir=/home/seanla/Projects/beijing_ancestor_mtbc/convertvcf\n",
    "$ strains=$convertVcfDir/unique_strains.txt\n",
    "$ convertVcf=$convertVcfDir/convertvcf.py\n",
    "$ phymlDir=$outputPrefix/../phyml\n",
    "$ mkdir -p $outputPrefix/..\n",
    "$ mkdir -p $phymlDir\n",
    "$ python $convertVcf -i $inputDir -o $outputPrefix -s $strains -r \"(\\d|\\D)*_noresi.vcf$\" -f -p\n",
    "$ cp $outputPrefix.phy $phymlDir\n",
    "```\n",
    "\n",
    "The list of the 110 strains used are contained in the directory `snps/unique_samples.txt`. The PHYLIP file at `snps/run2.phy` indicated that this sets of SNPs contains 8831 sites.\n",
    "\n",
    "jModelTest was used to infer the best fit model of nucleotide substitution for our data set. This was done with the following commands:\n",
    "\n",
    "```\n",
    "$ run=run2\n",
    "$ prefix=/global/scratch/seanla/Data/MTBC/$run\n",
    "$ phylip=$prefix/snps/$run.phy\n",
    "$ jmodelDir=$prefix/jmodeltest\n",
    "$ output=$jmodelDir/jmodeltest-results.out\n",
    "$ jModelTest=/home/seanla/Software/jmodeltest-2.1.10/jModelTest.jar\n",
    "$ mkdir -p $jmodelDir\n",
    "$ java -jar $jModelTest -tr ${PBS_NUM_PPN} -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "An explanation of the parameters used:\n",
    "* `-AIC` indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* `-a` indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* `-f` indicates we included models with unequal base frequencies.\n",
    "* `-g 4` indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* `-i` indicates we included models with a proportion invariable sites.\n",
    "* `-H AIC` indicates we used the AIC information criterion for clustering search.\n",
    "\n",
    "jModelTest outputted the General Time Reversible model with rate variation among sites and invariable sites as the best fit model of the data. The results of this can be found at `jmodeltest/jmodeltest-results.out`.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+I+G         0.16    0.34    0.33    0.17    0.00    0.00      0.932   3.448   0.594   0.610   2.804   1.000    0.00    0.73\n",
    "\n",
    "```\n",
    "\n",
    "## Construction of the neighbor-joining tree ##\n",
    "MEGA was used to construct a phylogenetic tree using neighbor-joining with the following commands:\n",
    "\n",
    "```\n",
    "$ run=run2\n",
    "$ prefix=/global/scratch/seanla/Data/MTBC/$run\n",
    "$ outputDir=$prefix/mega\n",
    "$ input=$prefix/snps/$run.fasta\n",
    "$ config=$prefix/../mao/infer_NJ_nucleotide.mao\n",
    "$ output=$outputDir/${run}mega\n",
    "$ mega=/home/seanla/Software/mega/megacc\n",
    "$ mkdir -p $outputDir\n",
    "$ mega -a $config -d $input -o $output\n",
    "```\n",
    "\n",
    "The following parameters were set:\n",
    "* Nucleotide alignment (non-coding)\n",
    "* Test of phylogeny - bootstrap with 1000 pseudoreplicates\n",
    "* Substitution model - no. of differences, transitions and transversions\n",
    "* Rates among sites - uniform rates (this was set incorrectly; this should have been set to rate variation among sites as per the output of jModelTest)\n",
    "\n",
    "The consensus tree in Newick format can be found at `mega/run2mega_consensus.newick` or `mega/run2mega_consensus.nwk` (same file; different extensions).\n",
    "\n",
    "![Neighbor-joining tree](mega/run2mega_consensus.png)\n",
    "\n",
    "## Construction of the Bayesian consensus tree ##\n",
    "BEAST 2 was used to find the optimal distribution of trees. The XML file was created using BEAUti using the following parameters\n",
    "\n",
    "**Site model**\n",
    "* The substitution model was set to GTR with rates of base conversion set to 1.0 and estimated (with the exception of CT, which was the default option by BEAST)\n",
    "* The gamma category count was set to 4, as per the settings we used in jModelTest to find the best fit model of DNA evolution.\n",
    "* The shape parameter was set to 1.0 (this was set incorrectly; the proper value should have been 0.73 as per the output of jModelTest)\n",
    "* The proportion of invariant sites was set to 0.0\n",
    "\n",
    "**Clock model**\n",
    "* A strick clock model was used, with rate 1.0\n",
    "\n",
    "**Priors**\n",
    "* The tree prior was set to the Yule Model.\n",
    "* Birth rate was set to uniform.\n",
    "* Substitution parameters were set to Gamma distributed.\n",
    "\n",
    "** MCMC **\n",
    "* The chain length used was 1500000, which is the number of chains used in the paper \"T cell epitopes are evolutionarily hyperconserved\" when constructing their phylogenies.\n",
    "* Pre-burnin was set to 150000\n",
    "* Every 1000th tree was stored.\n",
    "\n",
    "The distribution of trees outputted by BEAST2 (found at `beast/run2.trees`) was then fed into TreeAnnotator to find the consensus tree, which can be found at `beast/run2beast.nexus`. The tree was then visualized using FigTree.\n",
    "\n",
    "![Bayesian consensus tree](beast/run2beast.nexus.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
