{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run 6 - 88 Strains #\n",
    "## 19 July 2016 ##\n",
    "### Converting VCF to FASTA and PHYLIP ###\n",
    "The VCF files of the strains given in the file `snps/sequences_88.txt` (provided by Fang Zhang by email) were converted to FASTA and PHYLIP formats using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run6\n",
    "$ vcfDir=/home/zhf615/TB_test/MALAWI/test/Beijing/results\n",
    "$ projectDir=/home/seanla/Projects/beijing_ancestor_mtbc\n",
    "$ convertVcf=$projectDir/convertvcf/convertvcf.py\n",
    "$ runDir=$projectDir/$run\n",
    "$ snpDir=$runDir/snps\n",
    "$ snpPrefix=$snpDir/$run\n",
    "$ strains=$runDir/sequences_88.txt\n",
    "$\n",
    "$ mkdir -p $snpDir\n",
    "$ python $convertVcf -i $vcfDir -o $snpPrefix -s $strains -r  \"(\\d|\\D)*_unresi_unrepeted_depthfilerted.vcf$\" -f -p\n",
    "```\n",
    "Conversion yielded sequences with 6222 sites each.\n",
    "\n",
    "### Finding the best fit model of DNA substitution using jModelTest ###\n",
    "The PHYLIP file `snps/run6.phy` outputted by `convertvcf.py` was then inputted into jModelTest to find the best fit model of DNA substitution using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run6\n",
    "$ \n",
    "$ prefix=/home/seanla/Projects/beijing_ancestor_mtbc/$run\n",
    "$ phylip=$prefix/snps/$run.phy\n",
    "$ jmodelDir=$prefix/jmodeltest\n",
    "$ output=$jmodelDir/jmodeltest-results.out\n",
    "$ jModelTest=/home/seanla/Software/jmodeltest-2.1.10/jModelTest.jar\n",
    "$\n",
    "$ mkdir -p $jmodelDir\n",
    "$ \n",
    "$ java -jar $jModelTest -tr ${PBS_NUM_PPN} -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "An explanation of the parameters used is as follows:\n",
    "* -AIC indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* -a indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* -f indicates we included models with unequal base frequencies.\n",
    "* -g 4 indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* -i indicates we included models with a proportion invariable sites.\n",
    "* -H AIC indicates we used the AIC information criterion for clustering search. (jModelTest indicated that this option has no effect for our dataset.)\n",
    "\n",
    "Results from jModelTest indicated that the best fit model of DNA evolution for our data was the General Time Reversible Model with rate variation among sites (GTR+G), with a gamma parameter of 3.74.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+G           0.19    0.31    0.30    0.19    0.00    0.00      1.214   3.649   0.487   0.722   3.363   1.000 N/A        3.74\n",
    "```\n",
    "\n",
    "### Inferring the Neighbor-joining Tree using MEGA ###\n",
    "\n",
    "The MEGA configuration file was first constructed using `megaproto` with the following settings:\n",
    "* Test of phylogeny was set to bootstrap method using 1000 pseudoreplicates.\n",
    "* The substitution model was set to No. of differences, transitions and transversions being included.\n",
    "* Rates among sites was gamma distributed with a gamma parameter of 3.74.\n",
    "* All other settings were left at their defaults.\n",
    "\n",
    "The program was run with the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run6\n",
    "$\n",
    "$ prefix=/home/seanla/Projects/beijing_ancestor_mtbc/$run\n",
    "$ outputDir=$prefix/nj\n",
    "$ input=$prefix/snps/$run.fasta\n",
    "$ config=$outputDir/infer_NJ_nucleotide.mao\n",
    "$ output=$outputDir/${run}mega-nj\n",
    "$ mega=/home/seanla/Software/mega/megacc\n",
    "$ \n",
    "$ mkdir -p $outputDir\n",
    "$ mega -a $config -d $input -o $output\n",
    "```\n",
    "\n",
    "The consensus tree outputted by MEGA can be found at `nj/ru6mega-nj.newick`.\n",
    "\n",
    "![Neighbor-joining tree](nj/run6mega-nj_consensus.png)\n",
    "\n",
    "### Inferring the optimal tree distribution using Bayesian Inference via BEAST 2 ###\n",
    "\n",
    "The BEAST XML configuration file was first constructed using `beauti` with the following settings:\n",
    "\n",
    "*** Site Model ***\n",
    "* The model of DNA substitution was set to GTR.\n",
    "* The gamma category count was set to 4.\n",
    "* The shape parameter was set to 3.74.\n",
    "* The substitution rates parameters were all left at their defaults, with rates of 1.0 and estimated (with the exception of CT, which was the default option). \n",
    "* Frequencies were estimated.\n",
    "\n",
    "*** Clock Model ***\n",
    "* The clock model used was strict, with a clock rate of 1.0.\n",
    "\n",
    "*** Priors ***\n",
    "* The prior tree used was the Yule Model.\n",
    "\n",
    "The following parameters were left at their defaults.\n",
    "* Birth rate was set to uniform.\n",
    "* Rates were set to gamma distributed.\n",
    "\n",
    "*** MCMC ***\n",
    "* Chain length used was 10 000 000.\n",
    "* Pre burnin was set to 0.\n",
    "* Initialization attempts was set to 10.\n",
    "* Sample from prior was not set.\n",
    "\n",
    "BEAST was run using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run6\n",
    "$ prefix=/home/seanla/Projects/beijing_ancestor_mtbc/$run/bayes\n",
    "$ input=$prefix/${run}beast.xml\n",
    "$ output=$prefix\n",
    "$ \n",
    "$ mkdir -p $output\n",
    "$ \n",
    "$ export LD_LIBRARY_PATH=/home/seanla/lib:$LD_LIBRARY_PATH\n",
    "$ module load java\n",
    "$ export BEAGLE_LIB=/global/software/beagle/beagle212/lib\n",
    "$ export PATH=/global/software/beast/beast241/bin:$PATH\n",
    "$\n",
    "$ beast -threads ${PBS_NUM_PPN} -prefix $output -overwrite -working $input\n",
    "```\n",
    "\n",
    "The distribution of trees outputted by BEAST 2 (`bayes/run6.trees`) was then given to TreeAnnotator to construct the consensus tree given in NEXUS format (`bayes/run6bayes-50limit.nexus`). In TreeAnnotator notably, branch lengths were  set to mean heights and the posterior probability limit was set to 0.5. The tree was then visualized using FigTree.\n",
    "\n",
    "![Bayesian consensus tree](bayes/run6bayes-50limit.nexus.png)\n",
    "\n",
    "### Inferring the maximum likelihood tree using RAxML ###\n",
    "The maximum likelihood tree was constructed using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run6\n",
    "$ dir=/home/seanla/Projects/beijing_ancestor_mtbc/$run\n",
    "$ input=$dir/snps/$run.phy\n",
    "$ outputdir=$dir/raxml\n",
    "$ output=${run}raxml\n",
    "$\n",
    "$ mkdir -p $outputdir\n",
    "$\n",
    "$ raxml -T ${PBS_NUM_PPN} -f a -x 61938 -p 889579 -# 1000 -k -m GTRGAMMA -s $input -w $outputdir -n $output\n",
    "```\n",
    "\n",
    "An explanation of the parameters are as such:\n",
    "* `-f a` indicates we performed rapid bootstrap analysis and search for best-scoring ML tree in one program run.\n",
    "* `-x 61938` indicates we turned on rapid bootstrapping with a random seed of 61938.\n",
    "* `-p 889575` indicates we specified a random seed of 889575 for the parsimony inferences. This is required for algorithms using some sort of randomization.\n",
    "* `-#` indicates we performed 1000 pseudoreplicates for bootstrapping.\n",
    "* `-k` specifies the bootstrapped trees are printed with branch lengths.\n",
    "* `-m GTRGAMMA` indicates we used GTR as our model of DNA substitution, with estimated shape parameter.\n",
    "\n",
    "The tree outputted by RAxML can be found at `raxml/RAxML_bipartitionsBranchLabels.run6raxml.newick`. This tree was then visualized using Dendroscope.\n",
    "\n",
    "![Maximum likelihood tree](raxml/RAxML_bipartitionsBranchLabels.run6raxml.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
