{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run 7 - 90 Samples #\n",
    "\n",
    "## 25 July 2016 ##\n",
    "\n",
    "This experiment used the VCF files located at `/home/zhf615/TB_test/MALAWI/test/Beijing/results` in the Breezy cluster at Westgrid. The 90 samples used can be found in the file `snps/sequences_90.txt`.\n",
    "\n",
    "### Conversion of VCF to FASTA/PHYLIP ###\n",
    "\n",
    "The VCF files were converted to FASTA and PHYLIP format using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run7\n",
    "$ vcfDir=/home/zhf615/TB_test/MALAWI/test/Beijing/results\n",
    "$ projectDir=/home/seanla/Projects/beijing_ancestor_mtbc\n",
    "$ convertVcf=$projectDir/convertvcf/convertvcf.py\n",
    "$ runDir=$projectDir/$run\n",
    "$ snpDir=$runDir/snps\n",
    "$ snpPrefix=$snpDir/$run\n",
    "$ strains=$runDir/snps/sequences_90.txt\n",
    "$\n",
    "$ mkdir -p $snpDir\n",
    "$ python $convertVcf -i $vcfDir -o $snpPrefix -s $strains -r  \"(\\d|\\D)*_unresi_unrepeted_depthfilerted.vcf$\" -f -p\n",
    "```\n",
    "\n",
    "The alignments in the FASTA and PHYLIP files featured 6403 sites and 1579 patterns.\n",
    "\n",
    "### Finding the best fit model of nucleotide substitution ###\n",
    "\n",
    "The best fit model of nucleotide substitution was found using jModelTest 2 with the following commands:\n",
    "\n",
    "```bash\n",
    "run=run7\n",
    "prefix=/home/seanla/Projects/beijing_ancestor_mtbc/run7\n",
    "phylip=$prefix/snps/$run.phy\n",
    "jmodelDir=$prefix/jmodeltest\n",
    "output=$jmodeldir/jmodeltest-results.out\n",
    "jmodeltest=/home/seanla/Software/jmodeltest-2.1.10/jModelTest.jar\n",
    "\n",
    "mkdir -p $jmodeldir\n",
    "java -jar $jmodeltest -tr ${PBS_NUM_PPN} -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "An explanation of the parameters used is as follows:\n",
    "* -AIC indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* -a indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* -f indicates we included models with unequal base frequencies.\n",
    "* -g 4 indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* -i indicates we included models with a proportion invariable sites.\n",
    "* -H AIC indicates we used the AIC information criterion for clustering search. (jModelTest indicated that this option has no effect for our dataset.)\n",
    "\n",
    "Results from jModelTest indicated that the best fit model of DNA evolution for our data was the General Time Reversible Model with rate variation among sites and invariant sites (GTR+I+G), with a gamma parameter of 4.00 and 0.00 proportion of invariant sites.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+I+G         0.19    0.31    0.30    0.19    0.00    0.00      1.199   3.675   0.472   0.719   3.352   1.000    0.00    4.00\n",
    "```\n",
    "\n",
    "## 26 July 2016 ##\n",
    "\n",
    "### Construction of neighbor-joining tree ###\n",
    "\n",
    "The neighbor-joining tree was constructed using MEGA. The MAO configuration file was created using `megaproto` with the following settings:\n",
    "* Test of phylogeny was set to bootstrap method using 1000 pseudoreplicates.\n",
    "* The substitution model was set to No. of differences, transitions and transversions being included.\n",
    "* Rates among sites was gamma distributed with invariant sites with a gamma parameter of 4.00.\n",
    "* All other settings were left at their defaults.\n",
    "\n",
    "MEGA was run using the following commands:\n",
    "\n",
    "```bash\n",
    "run=run7\n",
    "prefix=/home/seanla/Projects/beijing_ancestor_mtbc/run7\n",
    "outputdir=$prefix/nj\n",
    "output=$outputdir/${run}mega-nj\n",
    "config=$outputdir/infer_NJ_nucleotide.mao\n",
    "input=$prefix/snps/${run}.fasta\n",
    "mega=/home/seanla/Software/mega/megacc\n",
    "\n",
    "mkdir -p $outputdir\n",
    "$mega -a $config -d $input -o $output\n",
    "```\n",
    "\n",
    "The tree outputted by MEGA can be found at `nj/run7mega-nj.newick`.\n",
    "\n",
    "![Neighbor-joining tree](nj/run7mega-nj_consensus.png)\n",
    "\n",
    "### Construction of Maximum Likelihood tree ###\n",
    "\n",
    "The maximum likelihood tree was constructed using the program RAxML using the following commands:\n",
    "\n",
    "```bash\n",
    "run=run7\n",
    "prefix=/home/seanla/Projects/beijing_ancestor_mtbc/run7\n",
    "input=$prefix/snps/${run}.phy\n",
    "outputdir=$prefix/raxml\n",
    "output=${run}raxml\n",
    "mkdir -p $outputdir\n",
    "raxml -T ${PBS_NUM_PPN} -f a -x 61938 -p 889579 -# 1000 -k -m GTRGAMMA -s $input -w $outputdir -n $output\n",
    "```\n",
    "\n",
    "An explanation of the parameters used is as follows:\n",
    "* `-f a` indicates we performed rapid bootstrap analysis and search for best-scoring ML tree in one program run.\n",
    "* `-x 61938` indicates we turned on rapid bootstrapping with a random seed of 61938.\n",
    "* `-p 889575` indicates we specified a random seed of 889575 for the parsimony inferences. This is required for algorithms using some sort of randomization.\n",
    "* `-#` indicates we performed 1000 pseudoreplicates for bootstrapping.\n",
    "* `-k` specifies the bootstrapped trees are printed with branch lengths.\n",
    "* `-m GTRGAMMA` indicates we used GTR as our model of DNA substitution, with estimated shape parameter\n",
    "\n",
    "The tree outputted by RAxML can be found at `raxml/RAxML_bipartitionsBranchLabels.run7raxml.newick`.\n",
    "\n",
    "![Maximum likelihood tree](raxml/RAxML_bipartitionsBranchLabels.run7raxml.png)\n",
    "\n",
    "### Finding the optimal tree distribution using Bayesian Inference ###\n",
    "\n",
    "The BEAST XML configuration file was first constructed using `beauti` with the following settings:\n",
    "\n",
    "*** Site Model ***\n",
    "* The model of DNA substitution was set to GTR.\n",
    "* The gamma category count was set to 4.\n",
    "* The shape parameter was set to 4.00.\n",
    "* The proportion of invariant sites was set to 0.00.\n",
    "* The substitution rates parameters were all left at their defaults, with rates of 1.0 and estimated (with the exception of CT, which was the default option). \n",
    "* Frequencies were estimated.\n",
    "\n",
    "*** Clock Model ***\n",
    "* The clock model used was strict, with a clock rate of 1.0.\n",
    "\n",
    "*** Priors ***\n",
    "* The prior tree used was the Yule Model.\n",
    "\n",
    "The following parameters were left at their defaults.\n",
    "* Birth rate was set to uniform.\n",
    "* Rates were set to gamma distributed.\n",
    "\n",
    "*** MCMC ***\n",
    "* Chain length used was 10 000 000.\n",
    "* Pre burnin was set to 0.\n",
    "* Initialization attempts was set to 10.\n",
    "* Sample from prior was not set.\n",
    "\n",
    "BEAST was run using the following commands:\n",
    "\n",
    "```bash\n",
    "$ run=run7\n",
    "$ prefix=/home/seanla/Projects/beijing_ancestor_mtbc/$run/bayes\n",
    "$ input=$prefix/${run}beast.xml\n",
    "$ output=$prefix\n",
    "$ \n",
    "$ mkdir -p $output\n",
    "$ beast -threads ${PBS_NUM_PPN} -prefix $output -overwrite -working $input\n",
    "```\n",
    "\n",
    "The consensus tree was then constructed using TreeAnnotator, where the following parameters were set:\n",
    "* 10% of generations were discarded as burn-in\n",
    "* The posterior probability limit was set to 0.5.\n",
    "\n",
    "![Bayesian consensus tree](bayes/run7-bayes_consensus.nexus.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
