{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Beijing Lineage MTBC - Phylogenetic Run 3 ##\n",
    "\n",
    "### 30 June 2016 ###\n",
    "#### Input Data ####\n",
    "The input data for this run are VCF files containing the SNPs from 110 unique strains from ~150 samples from the Beijing Lineage, where SNPs from repetitive regions and antibiotic-resistance have been removed. The entire list of strains used in this run can be found in the `convertvcf` folder in the file `unique_strains.txt`.\n",
    "\n",
    "#### VCF Conversion #####\n",
    "The collection of VCF files were converted to FASTA file format with the tool `convertvcf.py` under the `convertvcf` folder with the following commands:\n",
    "\n",
    "```bash\n",
    "run=run3\n",
    "inputDir=/home/zhf615/TB_test/MALAWI/test/shuffled/results\n",
    "outputDir=/global/scratch/seanla/Data/MTBC/$run\n",
    "snpDir=$outputDir/snps\n",
    "snpPrefix=$snpDir/$run\n",
    "convertVcfDir=/home/seanla/Projects/beijing_ancestor_mtbc/convertvcf\n",
    "strains=$convertVcfDir/unique_strains.txt\n",
    "convertVcf=$convertVcfDir/convertvcf.py\n",
    "\n",
    "mkdir -p $snpDir\n",
    "\n",
    "python $convertVcf -i $inputDir -o $snpPrefix -s $strains -r \"(\\d|\\D)*_unresi_unrepeted.vcf$\" -f -p\n",
    "```\n",
    "\n",
    "Where `(\\d|\\D)*_unresi_unrepeted.vcf$` is the regular expression for the appropriate VCF files. Processing revealed that there are 7806 sites (i.e. the length of each strain sequence is 7806 bases).\n",
    "\n",
    "#### jModelTest ####\n",
    "jModelTest was used to infer the best-fit model of nucleotide evolution using the Akaike information criterion with the following commands:\n",
    "\n",
    "```bash\n",
    "run=run3\n",
    "prefix=/global/scratch/seanla/Data/MTBC/$run\n",
    "phylip=$prefix/snps/$run.phy\n",
    "jmodelDir=$prefix/jmodeltest\n",
    "output=$jmodelDir/jmodeltest-results.out\n",
    "jModelTest=/home/seanla/Software/jmodeltest-2.1.10/jModelTest.jar\n",
    "\n",
    "mkdir -p $jmodelDir\n",
    "\n",
    "java -jar $jModelTest -tr ${PBS_NUM_PPN} -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "An explanation of the parameters used:\n",
    "* `-AIC` indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* `-a` indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* `-f` indicates we included models with unequal base frequencies.\n",
    "* `-g 4` indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* `-i` indicates we included models with a proportion invariable sites.\n",
    "* `-H AIC` indicates we used the AIC information criterion for clustering search. (jModelTest indicated that this option has no effect for our dataset.)\n",
    "\n",
    "jModelTest ouputted GTR+I+G (General Time Reversible model with invariable sites and rate variation among sites) as the best fit model for our data.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+I+G         0.18    0.32    0.31    0.18    0.00    0.00      0.915   3.079   0.432   0.566   2.729   1.000    0.00    2.16\n",
    "```\n",
    "\n",
    "#### Neighbor Joining ####\n",
    "To construct the neighbor joining tree, MEGA was used with the following parameters:\n",
    "* Nucleotide alignment (non-coding)\n",
    "* Test of phylogeny - bootstrap with 1000 pseudoreplicates\n",
    "* Substitution model - no. of differences, transitions and transversions\n",
    "* Rates among sites - gamma distributed with invariant sites, gamma parameter of 0\n",
    "* Gaps/Missing Data Treatment was set to deletion\n",
    "\n",
    "#### Maximum Likelihood ####\n",
    "To construct the Maximum Likelihood tree, MEGA was used with the following parameters:\n",
    "* The data to analyze was set to nucleotide alignment (non-coding).\n",
    "* Test of phylogeny was set to bootstrap method with 1000 replications.\n",
    "* The substitution model was set to GTR.\n",
    "* Rates among sites was set to Gamma Distributed with Invariant Sites (G+I)\n",
    "* The number of discrete gamma categories was set to 4.\n",
    "* Gaps/Missing Data were treated as complete deletion.\n",
    "* The ML heuristic method used was Nearest-Neighbor-Interchange (NNI)\n",
    "* The initial tree for ML was created using NJ/BioNJ (Neighbor-joining?)\n",
    "* No branch swap filters were used.\n",
    "\n",
    "#### Bayesian Inference ####\n",
    "To construct the Bayesian Inference tree, BEAST 2 was used with the following parameters:\n",
    "\n",
    "##### Site Model #####\n",
    "* The site model was set to gamma with a gamma category count of 4.\n",
    "* The shape of the gamma site model was set to 1.0\n",
    "* The proportion invariant of the gamma site model was set to 0.0.\n",
    "* The model of nucleotide substitution was set to GTR.\n",
    "* Rate for each nucleotide substitution was set to 1.0.\n",
    "* Each nucleotide substitution rate was estimated with the exception of CT.\n",
    "* Frequences were estimated as well.\n",
    "\n",
    "##### Clock Model #####\n",
    "* The clock model was set to strict with a rate of 1.0.\n",
    "\n",
    "##### Priors #####\n",
    "* The tree model was set to Yule.\n",
    "* The birth rate was set to uniform.\n",
    "* Each nucleotide rate prior was set to gamma.\n",
    "\n",
    "##### MCMC #####\n",
    "* 10 000 000 chains were used with 0 pre-burnin (not the same as burn-in).\n",
    "* We logged the tree distribution every 1000 chains.\n",
    "* We did not sample from prior."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
