{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Sunday, 19 June 2016**\n",
    "\n",
    "VCF files containing the SNPs from the Beijing lineage from the folder `/home/zhf615/TB_test/MALAWI/test/shuffled/results` were converted to the FASTA, NEXUS, and PHYLIP formats using the tool `convertvcf.py`. VCF files in the format `*_noresi.vcf` in each folder were selected for the tests.\n",
    "\n",
    "The following commands were used:\n",
    "\n",
    "```\n",
    "$ input=/home/zhf615/TB_test/MALAWI/test/shuffled/results\n",
    "$ prefix=/global/scratch/seanla/Data/MTBC/run1/snps\n",
    "$ output=$prefix/run1snps\n",
    "$ cd /home/seanla/Projects/beijing_ancestor_mtbc/convertvcf/\n",
    "$ python convertvcf.py -i $input -o $output -r \"(\\d|\\D)*_noresi.vcf$\" -f -n -p\n",
    "$ cd $output\n",
    "$ cp run1snp.phy ../phyml/\n",
    "```\n",
    "\n",
    "The PHYLIP file indicated that the length of each sequence is 9771.\n",
    "\n",
    "## Finding the best fit model ##\n",
    "jModelTest 2.1.10 was used on the PHYLIP file to infer the best fit model of DNA substitution with the following commands:\n",
    "\n",
    "```\n",
    "$ prefix=/global/scratch/seanla/Data/MTBC/run1\n",
    "$ phylip=$prefix/snps/run1snps.phy\n",
    "$ output=$prefix/jmodeltest/run1jmt.out\n",
    "$ cd /home/seanla/Software/jmodeltest-2.1.10\n",
    "$ java -jar jModelTest.jar -d $phylip -o $output -AIC -a -f -g 4 -i -H AIC\n",
    "```\n",
    "\n",
    "Here is an explanation of the parameters used:\n",
    "* `-AIC` indicates we used the Akaike information criterion to infer the best fit model.\n",
    "* `-a` indicates we estimated the model-averaged phylogeny for each active criterion.\n",
    "* `-f` indicates we included models with unequal base frequencies.\n",
    "* `-g 4` indicates we included models with rate variation among sites and sets the number of categories to 4.\n",
    "* `-i` indicates we included models with a proportion invariable sites.\n",
    "* `-H AIC` indicates we used the AIC information criterion for clustering search.\n",
    "\n",
    "The output of jModelTest indicated the best fit model for the Beijing lineage SNP data was GTR+G, or the General Time Reversible model with rate variation among sites.\n",
    "\n",
    "```\n",
    "::Best Models::\n",
    "\n",
    "        Model           f(a)    f(c)    f(g)    f(t)    kappa   titv    Ra      Rb      Rc      Rd      Re      Rf      pInv    gamma\n",
    "----------------------------------------------------------------------------------------------------------------------------------------\n",
    "AIC     GTR+G           0.16    0.34    0.33    0.17    0.00    0.00      0.920   3.380   0.634   0.618   2.859   1.000 N/A        0.60\n",
    "\n",
    "```\n",
    "\n",
    "Next, PhyML 3.1 was used to infer the best phylogenetic tree using a maximum likelihood approach with the following commands:\n",
    "\n",
    "```\n",
    "$ input=/global/scratch/seanla/Data/MTBC/run1/phyml/run1snps.phy\n",
    "$ cd /home/seanla/Software/PhyML-3.1\n",
    "$ ./PhyML-3.1_linux64 -i $input -q -b 1000 -m GTR --no_memory_check\n",
    "```\n",
    "\n",
    "Here is an explanation of the parameters used:\n",
    "* `-q` indicates the input data is in sequential form.\n",
    "* `-b 1000` indicates bootstrap was performed with 1000 pseudoreplicates.\n",
    "* `-m GTR` indicates the model used was GTR.\n",
    "* `--no_memory_check` indicates that the program was run without check of sufficient memory.\n",
    "\n",
    "jModelTest indicated the best fit model includes rate variation among sites - PhyML implements this by automatically adjusting the gamma correction value by default. However, there is a `--free_rates` options that is an alternative to the discrete gamma model that may provide more accurate models. This is a point of discussion for the future."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Monday, 20 June 2016**\n",
    "\n",
    "## Bayesian Inference ##\n",
    "BEAST v1.8.3 was used to infer the optimal phylogenetic tree distribution using the Bayesian Inference method. The FASTA file `run1snps.fasta` was used as input. The BEAST XML file was created using the program BEAUti. The following parameters were changed:\n",
    "* Substitution model - changed to GTR, with site heterogeneity model Gamma with 4 categories.\n",
    "* Ancestral State Reconstructions - \"Reconstruct states at ancestor\" setting was activated and set to the tree root.\n",
    "* Length of chain - set to 1500000\n",
    "\n",
    "The remainder of the settings were left at their default values.\n",
    "\n",
    "MEGA was used to infer the optimal phylogenetic tree using neighbor-joining. The following parameters were set:\n",
    "* Nucleotide alignment (non-coding)\n",
    "* Test of phylogeny - bootstrap with 1000 pseudoreplicates\n",
    "* Substitution model - no. of differences, transitions and transversions\n",
    "* Rates among sites - uniform rates\n",
    "\n",
    "![Bayesian inference consensus tree](beast/run1bayes.nexus.png)\n",
    "\n",
    "MEGA was run using the following commands:\n",
    "\n",
    "```\n",
    "$ prefix=/global/scratch/seanla/Data/MTBC/run1\n",
    "$ input=$prefix/snps/run1snps.fasta\n",
    "$ config=$prefix/mega/infer_NJ_nucleotide.mao\n",
    "$ output=$prefix/mega/run1mega\n",
    "$ cd /home/seanla/Software/mega\n",
    "$ ./megacc -a $config -d $input -o $output\n",
    "```\n",
    "\n",
    "![Neighbor joining tree with bootstrap values](mega/run1mega_consensus.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
